src/Surfnet/StepupRa/RaBundle/EventListener/InconsistentStateListener.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+/**
+ * Copyright 2016 SURFnet B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Surfnet\StepupRa\RaBundle\EventListener;
+
+use Psr\Log\LoggerInterface;
+use Surfnet\StepupRa\RaBundle\Exception\InconsistentStateException;
+use Symfony\Component\HttpKernel\Event\GetResponseForExceptionEvent;
+
+class InconsistentStateListener
+{
+    /**
+     * @var LoggerInterface
+     */
+    private $logger;
+
+    public function __construct(LoggerInterface $logger)
+    {
+        $this->logger = $logger;
+    }
+
+    public function onKernelException(GetResponseForExceptionEvent $event)
+    {
+        $exception = $event->getException();
+
+        if ($exception instanceof InconsistentStateException) {
```

Comment:
it's always nicer to check if you can handle it, if not return early. Smaller if, faster flow.